In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("weather_vn_2125.csv")

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 113148 entries, 0 to 113147
Data columns (total 9 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   location             113148 non-null  object 
 1   date                 113148 non-null  object 
 2   weather_code         113148 non-null  int64  
 3   temp_max             113148 non-null  float64
 4   temp_min             113148 non-null  float64
 5   temp                 113148 non-null  float64
 6   wind_speed           113148 non-null  float64
 7   humidity             113148 non-null  float64
 8   weather_description  113148 non-null  object 
dtypes: float64(5), int64(1), object(3)
memory usage: 7.8+ MB


## Data description
### Overview: 
The dataset consists of Vietnam's 63 provinces/cities (before merger of provinces) weather, collected daily from 2021-01-01 to 2025-12-01. This dataset provide regular information of weather for touristic purpose: temperature, wind speed, humidity, weather description
### Source: 
Data is collected from Open Meteo Historical Weather API
https://open-meteo.com/en/docs/historical-weather-api
### Data Statistics:
- 113147 rows, 7 columns, each row include a province/city's weather data in one day
Columns descripton:
- location(object): province / city name
- date(object): date
- weather_code(int64): weather code according to WMO code table 4677, each code stand for a weather condition
- temp, temp_max, temp_min: respectively average temperature, maximum temperature, minimum temperature
- wind_speed: average wind speed
- humidity: relative humidity
- weather description: overall weather condition inferred from weather code